server/hotrod/src/main/java/org/infinispan/server/hotrod/ServerAddress.java

```diff
@@ -0,0 +1,102 @@
+package org.infinispan.server.hotrod;
+
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.remoting.transport.Address;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A Hot Rod server address
+ *
+ * @author Galder Zamarreño
+ * @since 5.1
+ */
+public class ServerAddress implements Address {
+   private final String host;
+   private final int port;
+
+   public ServerAddress(String host, int port) {
+      this.host = Objects.requireNonNull(host);
+      this.port = port;
+   }
+
+   @Override
+   public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+
+      ServerAddress that = (ServerAddress) o;
+
+      if (port != that.port) return false;
+      return host.equals(that.host);
+
+   }
+
+//   // IMPORTANT NOTE: Hot Rod protocol agrees to this calculation for a node
+//   // address hash code calculation, so any changes to the implementation
+//   // require modification of the protocol.
+//   override def hashCode() = Arrays.hashCode(
+//      "%s:%d".format(host, port).getBytes(UTF8))
```

Comment:
This was actually something I need to look into, thanks for reminding me.  This code was here before like this but commented out.  It talks about the hashCode requiring a specific implementation, which the old code didn't do.  I wanted to make sure that was alright before I removed it.